src/server/main_service.cc

```diff
@@ -2161,6 +2161,18 @@ void Service::PubsubPatterns(ConnectionContext* cntx) {
   (*cntx)->SendLong(pattern_count);
 }
 
+void Service::PubsubNumSub(const vector<string_view>& channels, ConnectionContext* cntx) {
+  vector<std::pair<std::string, long>> pubsub_num_array;
+  pubsub_num_array.reserve(channels.size() * 2);
+
+  for (auto channel : channels) {
+    pubsub_num_array.emplace_back(
+        channel, ServerState::tlocal()->channel_store()->FetchSubscribers(channel).size());
+  }
+
+  (*cntx)->SendNumSubArray(pubsub_num_array);
```

Comment:
To clarify, the logic of `PubsubNumSun()` should be here instead of in `ReplyBuilder`

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/0abca50381ec31ee7332cc95617af1ec9dad1eae